lighthouse-core/report/html/renderer/report-renderer.js

```diff
@@ -195,6 +195,7 @@ class ReportRenderer {
     const detailsRenderer = new DetailsRenderer(this._dom);
     const categoryRenderer = new CategoryRenderer(this._dom, detailsRenderer);
     categoryRenderer.setTemplateContext(this._templateContext);
+    categoryRenderer.setLighthouseReport(report);
```

Comment:
the perf category renderer and a separate pr #10716 both need some data from the LHR to render, so let's put the report on the details renderer

done via `setLighthouseReport` so that the property will be optional and existing usages won't break (i.e. progressive enhancement)